[1.11] dcos-l4lb routes traffic until the first failed health check

urbanserj opened the pull request:
JIRA: https://jira.mesosphere.com/browse/DCOS_OSS-1954
mesos_state PR: https://github.com/dcos/mesos_state/pull/39

Fabs commented:
Have you considered adding a small unit test to exercise the health check behavior change? This looks to me that this is a backports, but if you consider adding it here that would be cool to have that on master too :D.

urbanserj commented:
Hey @Fabs, thank you for the good suggestion. I've been waiting for a dc/os build with patched mesos [1]. I've added some tests to l4lb, please take a look. In DC/OS 1.12, we moved the logic to parse and it already had some tests for `healthy` task status [2] and we added yet another test into master branch for this specific case [3].

[1] https://github.com/dcos/dcos/pull/3720
[2] https://github.com/dcos/dcos-net/blob/master/apps/dcos_net/test/dcos_net_mesos_listener_tests.erl
[3] https://github.com/dcos/dcos-net/commit/6d9eb3f8ca7608098fd3b7ce258898cee090e51e#diff-1f5ea808f96bf11fede1865d14067762R120

urbanserj commented:
@GoelDeepak request for review
